lxd/devices.go

```diff
@@ -393,30 +453,97 @@ func deviceTaskBalance(s *state.State) {
 			continue
 		}
 
+		fillFixedInstances := func(targetCpuPool []int64, targetCpuNum int, loadBalancing bool) {
```

Comment:
It doesn't seem appropriate to define this function inside the for loop, can we make it its own discrete standalone function, importantly with a comment explaining what it does?